execution_engine/src/storage/global_state/mod.rs

```diff
@@ -47,7 +49,7 @@ pub enum CommitResult {
     Serialization(bytesrepr::Error),
 }
 
-#[derive(Debug)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct ReadTrieResult {
     pub trie_key: Blake2bHash,
     pub maybe_trie: Option<Trie<Key, StoredValue>>,
```

Comment:
@EdHastingsCasperLabs I wrote this prior to understanding how the fetcher works.  I need to know when I ask for a `trie_key` if I got the right value for that key.  If I got the wrong value, we should disconnect and if I got no value we should query another peer.

I'm thinking I can just use the `fetcher` Request/Response types instead of what I've done here.

Reply:
T could be Trie<Key, StoredValue> in which case the existing event set up for the generic fetcher should work as is.

Reply:
Return an `Option<Trie<Key, StoredValue>>` and put that directly into an `Event::GetFromStorageResult`.

Reply:
Fixed in https://github.com/CasperLabs/casper-node/pull/768/commits/0b5a8caa002d16b36f742017215a27919ce9314f